### PR TITLE
Add new Corkscrew Roller Coaster track pieces

### PR DESCRIPTION
**Pull request by mixiate**
This adds new pieces to the Corkscrew Roller Coaster.

Included are:
- Small flat to steep
- Vertical slopes
- Vertical turns
- Banked slope transitions
- Small banked sloped curves
- Medium banked sloped curves
- Large sloped curves
- Large banked sloped curves
- Small banked to unbanked curves
- Large corkscrews
- Quarter loops
- Medium half loops
- Large half loops
- Barrel rolls
- Zero g rolls
- Large zero g rolls

allcorkscrewpieces

The large sloped curves, small banked to unbanked curves, large corkscrews, medium and large half loops are made available to the corkscrew roller coaster. The banked sloped curves are available for trains with those sprites. All other pieces are available through cheats only.

I've done my best to make all the pieces function correctly. I've tested with the RCT 1 and 2 trains, [spaceks modern corkscrew trains](https://github.com/spacek531/custom-rides/raw/master/Modern%20Corkscrew%20CTR.zip), and a test train I made that has roughly the same dimensions as the RCT 2 train.
Some compromises were made because it was quite difficult to get the corkscrew train to go through some of these pieces without glitching.

I've also added the code to change the version and make pre existing pieces invisible.

### CODE REVIEW

**Comment by SpartanFrederic104** [comment]
Fred's Workbench 2024-10-08 22-41-59
Tested these new track pieces and so far they look insanely nice. Large half corkscrews and larger loops are always welcomed, especially in a coaster type that tends to be very intense partly due to the RCT2 trains giving it a +10% intensity boost (the RCT1 trains don't do this) and the relatively high G-forces from entering regular corkscrews.

Fred's Workbench 2024-10-08 22-43-16
If I had any issues with this, it would be a relatively small issue with the lighting/shading at the start/end of the medium zero-G roll piece in this angle pictured above (it instantly goes from bright to a slightly darkened hue as soon as the zero-G roll starts/ends). Other than that, the new track pieces look great and I can't wait to see these new track pieces in action in the official builds soon enough.

**Comment by xrgtmx** [comment]
Needs changelog, network, and park tags.

**Comment by mixiate** [comment]
Thanks for testing out these pieces again!

I haven't commited it yet but I tried to lighten the zero g roll a little bit without completely removing the shadow.

newzerogroll

Let me know what you think.

**Comment by SpartanFrederic104** [comment]
That looks better now.

**Comment by Gymnasiast** [comment]
I’ll look at this one after the Wooden RC loops PR has been merged.

**Comment by Gymnasiast** [comment]
Could you rebase this one?

**Comment by mixiate** [comment]
I've rebased this. Apologies for the several pushes, i'm not getting these warnings/errors when compiling myself for some reason. I will see if I can do anything about that.

I hope it's okay but I also added a commit to make the curves available for the lay down coaster, and added it to the list for invisible pieces. Sorry I didn't catch this before.

File: src/openrct2/paint/track/coaster/CorkscrewRollerCoaster.cpp [hunk 1]
```diff
@@ -11046,6 +11046,8353 @@ static void LayDownRCTrackHalfLoopUninvertedDown(
...
+                        PaintSegment::leftCorner, PaintSegment::centre, PaintSegment::topLeftSide,
+                        PaintSegment::bottomLeftSide),
+                    direction),
+                0xFFFF, 0);
+            PaintUtilSetGeneralSupportHeight(session, height + 64);
+            break;
+    }
+}
+
+static void CorkscrewRCTrack90DegUp(
```
Comment (Gymnasiast) [inline src/openrct2/paint/track/coaster/CorkscrewRollerCoaster.cpp:11552]: There are already sprites for this on the Compact Inverted Coaster. Is there any specific reason you didn’t reuse these?

Comment (mixiate) [reply src/openrct2/paint/track/coaster/CorkscrewRollerCoaster.cpp:11552]: Ah I didn't even realise. I could use it if you'd like. I'd have to rerender a few pieces to join it correctly.

**Comment by karst** [comment]
Idem on the conversion code I mentioned in another one of your PRs.

**Comment by mixiate** [comment]
They both already have it don't they? Does it not work?

**Comment by mixiate** [comment]
This should be okay now if you wanted to take another look at it.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**